tree/treeplayer/test/treeprocs/treeprocessors.cxx

```diff
@@ -56,6 +59,42 @@ void DeleteFiles(const std::vector<std::string> &filenames)
       gSystem->Unlink(f.c_str());
 }
 
+class FilesRAII {
+   std::vector<std::string> fFileNames;
+
+public:
+   FilesRAII(const std::vector<std::string> &treenames, const std::vector<std::string> &filenames, int nEvts = 10)
+      : fFileNames(filenames)
+   {
+      WriteFiles(treenames, filenames, nEvts);
+   }
+   ~FilesRAII() { DeleteFiles(fFileNames); }
+};
+
+// See issue #15425
+TEST(TreeProcessorMP, moreWorkersThanEvents)
+{
+   auto func = [](TTreeReader &r){
+      int n=0;
+      while(r.Next()) n++;
+      auto par = new TParameter<int>("n", n);
+      return par;
+   };
+
+   std::vector<std::string> files = {"f_moreWorkersThanEvents.root"};
+   std::vector<std::string> trees = {"t"};
+
+   FilesRAII fr(trees, files, 2);
+
+   ROOT::TestSupport::CheckDiagsRAII checkDiag;
+   checkDiag.requiredDiag(kError, "TTreeProcessorMP::HandlePoolCode", "[E][C] a worker encountered an error: [S2]: could not set TTreeReader to range 2 1", false);
```

Comment:
This error is quite too dense to be understandable.
> could not set TTreeReader to range 2 1

First off I would try to make it clear that `2 1` is something like `begin=2, end=1`. Then In this case it's understandable because begin > end does not make sense.

But in general, what is the error telling to the user? Why did it happen that the entry range begin was set to a value greater than the end in the first place? I guess from how the test is constructed that a more sensible error/warning should inform the user that there were more processes than available entry ranges, so an invalid entry range was constructed in order to avoid reading the same entries more than once.

Reply:
I've noticed that this error got removed in the latest commit. This is even more confusing tbh. Will the code actually trigger this error or not?

Maybe the new check for `start != fTree->GetEntries()` will simply not even create tasks that would have invalid ranges, so that's why the error is not there anymore. Is that it?